cmd/root.go

```diff
@@ -183,23 +183,22 @@ func CreateSystem(shutdownCallback context.CancelFunc) *core.System {
 	httpServerInstance := httpEngine.New(shutdownCallback, cryptoInstance)
 	jsonld := jsonld.NewJSONLDInstance()
 	storageInstance := storage.New()
-	didStore := didstore.New(storageInstance.GetProvider(vdr.ModuleName))
+	didResolver := &didservice.DIDResolverRouter{}
```

Comment:
After VDR's `Configure()` has been called, the DID resolvers are ready for work. But that's kinda by chance, since it could also just be after `Start()`. So we should do the following:

- Alter start sequence (`RegisterEngine()`) from `VCR, VDR, Network` to `VDR, VCR, Network` (doesn't actually matter, but VCR depends on VDR so VDR should be started first).
- Return a more explicit error than "unsupported DID method" in case no DID resolvers are registered on the router, indicating a programmer error.

Reply:
Network configure depends on the resolvers. Technically it could be solved by placing the VDR one place up in the startup sequence. Add a comment line to make sure that nobody changes it again.

But still it's a bit spaghetti. Could improve with the other DID method and change in storage....

Reply:
It already is like that: VDR and VCR are started before network is. Problem is; VDR depends on network (ambassador) and network on VDR (discovering nodes). This dependency will stay whatever we do, since is sent both up and down. We can make it a bit more convenient to work with (e.g. by decoupling using events for ambassador stuff), but the runtime dependency will stay.